main.go

```diff
@@ -0,0 +1,168 @@
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
+	var enableLeaderElection bool
+	var probeAddr string
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     fmt.Sprintf("%s:%d", metricsHost, metricsPort),
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "29fea974.configurealertmanager.managed.openshift.io",
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
+	}
+
+	// Print configuration info
+	printVersion()
+	if err := operatorconfig.SetIsFedramp(); err != nil {
+		setupLog.Error(err, "failed to get fedramp value")
+		os.Exit(1)
+	}
+	if operatorconfig.IsFedramp() {
+		setupLog.Info("running in fedramp environment.")
+	}
+
+	err = leader.Become(context.TODO(), "configure-alertmanager-operator-lock")
+	if err != nil {
+		setupLog.Error(err, "Failed to retry for leader lock")
+		os.Exit(1)
+	}
+
+	if err = (&controllers.SecretReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "Secret")
+		os.Exit(1)
+	}
+	//+kubebuilder:scaffold:builder
+
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+		setupLog.Error(err, "unable to set up ready check")
+		os.Exit(1)
+	}
+
+	// Create Service object to expose the metrics port.
+	s, svcerr := operatormetrics.GenerateService(8080, "metrics")
+	if svcerr != nil {
+		log.Error(err, "Error generating metrics service object.")
+		panic("Ensure that operator is running in a cluster and not in a local development environment.")
+	} else {
+		log.Info("Generated metrics service object")
+	}
+
+	sm := operatormetrics.GenerateServiceMonitor(s)
+	err = mgr.GetClient().Create(context.TODO(), s)
+	if err != nil {
+		log.Error(err, "error creating metrics Service")
+	} else {
+		log.Info("Created Service")
+		err = mgr.GetClient().Create(context.TODO(), sm)
+		if err != nil {
+			log.Error(err, "error creating metrics ServiceMonitor")
+		} else {
+			log.Info("Created ServiceMonitor")
+		}
+	}
+
+	log.Info("Starting prometheus metrics.")
+	if err := operatormetrics.StartMetrics(); err != nil {
+		log.Error(err, "Failed to start metrics service")
+		os.Exit(1)
+	}
+
```

Comment:
```

	// Check for fedramp clusters
	err = operatorconfig.SetIsFedramp()
	if err != nil {
		log.Error(err, "Failed to get fedramp value")
		os.Exit(1)
	}
	if operatorconfig.IsFedramp() {
		log.Info("Running in fedramp environment.")
	}
```

This block to check the fedRamp value is missed before we start the manager

Reply:
Its been moved higher in the code.

Reply:
overlooked... please ignore this one. Pete.